From 7ca999911f6f471f3e49a3ed34d6f42eddea3c59 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 19 Jan 2022 08:45:59 +0200
Subject: [PATCH] [SW-58071] common: move binary compilation device specific

Currently, ROM and RAM binaries are compiled for greco and gaudi2
devices however RAM binary is not required for gaudi3. This patch
move binary selection device specific.

Change-Id: I2499135111e1c73d87af075366034cb70c54fce0
---

diff --git a/Makefile b/Makefile
index a364a24..8821593 100644
--- a/Makefile
+++ b/Makefile
@@ -65,8 +65,7 @@
 
 .PHONY: all libsb libellipsys sbtool
 all: libsb libellipsys
-	@$(MAKE) arcboot_rom.bin
-	@$(MAKE) arcboot_ram.bin
+	@$(MAKE) $(BINARIES)
 
 libsb libellipsys:
 	@+$(MAKE) -C sdk $@ SB_SIGN_MODE=rsa_fixed > /dev/null
diff --git a/arch/gaudi2/arc/Makefile b/arch/gaudi2/arc/Makefile
index 1c6d192..ae139ef 100644
--- a/arch/gaudi2/arc/Makefile
+++ b/arch/gaudi2/arc/Makefile
@@ -1 +1,3 @@
 LDFLAGS += -static
+
+BINARIES = arcboot_rom.bin arcboot_ram.bin
diff --git a/arch/greco/arc_elf32/Makefile b/arch/greco/arc_elf32/Makefile
index e69de29..668dfae 100644
--- a/arch/greco/arc_elf32/Makefile
+++ b/arch/greco/arc_elf32/Makefile
@@ -0,0 +1 @@
+BINARIES = arcboot_rom.bin arcboot_ram.bin
